### DISCUSSION TITLE
set-passaccount for adding custom properties

### DISCUSSION THREAD

**Discussion by ELANDJEA** [pspete/psPAS#451] [Q&A]
using the PSPAS modules, I try to add/replace/remove custom properties.
However, I can't get it to work with adding a custom propertie; I constantly get a "Invoke-PASRestMethod : [400] Invalid Input Request. Reason: Object reference not set to an instance of an object." error.

for instance....
base account:
![afbeelding](https://user-images.githubusercontent.com/70526758/216961308-d0d8d0b8-a607-4e00-a7ca-885789039488.png)

when I use operator add or replace it works ok if the propertie is already there:
![afbeelding](https://user-images.githubusercontent.com/70526758/216961547-15b2a407-7c3c-4c82-bdfa-96d768f3f059.png)

when I after that remove the propertie and try to re-add it, it fails:
![afbeelding](https://user-images.githubusercontent.com/70526758/216961717-a8d9f989-5f68-48a6-9d5b-36a94e2c7ed3.png)

what am I doing wrong to get the custom propertie added via PSPAS?
in above example  of updating, I added the propertie manually via the PVWA by editing the custom field in the account properties.

note:
EPV 12.2
PVWA 12.2
and custom property "newdescription" added and available via file categories within PAClient

**Comment by pspete** [comment]
Unfortunately not something that is able to be replicated here, could it be an issue in your local environment or platform configuration?

![image](https://user-images.githubusercontent.com/10600090/222577966-d8ab772f-2809-45d4-9454-cb774e752b8a.png)

Consider following the [troubleshooting / manual api testing instructions](https://pspas.pspete.dev/docs/troubleshooting/#manual-api-command-testing) with the example code provided here to see if the same issue happens when not using the psPAS module.

```powershell
#after New-PASSession
$ThisSession = Get-PASSession

$AccountID = 21_4

$Method = "PATCH"
$UrlPath = "api/Accounts/$AccountID"

$Body = ConvertTo-Json @(@{"op" = "add";"path" = "newdescription";"value" = "your new description"})

$Request = @{
    "Method"      = $Method
    "Uri"         = "$($ThisSession.BaseUri)/$UrlPath"
    "Body"        = $Body
    "WebSession"  = $ThisSession.WebSession
    "ContentType" = "application/json"
}

Invoke-RestMethod @Request
```
Hope this helps
